shared/teams/team/member/add-to-channels.tsx

```diff
@@ -186,96 +189,224 @@ const AddToChannels = (props: Props) => {
       noScrollView={true}
       onClose={onCancel}
     >
-      <Kb.Box2 direction="vertical" fullWidth={true} style={Styles.globalStyles.flexOne}>
-        <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.searchFilterContainer}>
-          <Kb.SearchFilter
-            placeholderText={`Search ${channelMetasAll.length} ${pluralize(
-              'channel',
-              channelMetasAll.length
-            )}`}
-            icon="iconfont-search"
-            onChange={setFilter}
-            size="full-width"
-            onFocus={() => setFiltering(true)}
-            onBlur={() => setFiltering(false)}
-          />
-        </Kb.Box2>
-        <Kb.Box2 direction="vertical" style={styles.listContainer} fullWidth={true}>
-          <Kb.List2
-            items={items}
-            renderItem={renderItem}
-            itemHeight={
-              Styles.isMobile ? {height: 48, type: 'fixed'} : {sizeType: 'Large', type: 'fixedListItem2Auto'}
-            }
-          />
+      {loadingChannels ? (
+        <Kb.ProgressIndicator type="Huge" />
+      ) : (
+        <Kb.Box2 direction="vertical" fullWidth={true} style={Styles.globalStyles.flexOne}>
+          <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.searchFilterContainer}>
+            <Kb.SearchFilter
+              placeholderText={`Search ${channelMetasAll.length} ${pluralize(
+                'channel',
+                channelMetasAll.length
+              )}`}
+              icon="iconfont-search"
+              onChange={setFilter}
+              size="full-width"
+              hotkey="f"
+              onFocus={() => setFiltering(true)}
+              onBlur={() => setFiltering(false)}
+            />
+          </Kb.Box2>
+          <Kb.Box2 direction="vertical" style={styles.listContainer} fullWidth={true}>
+            <Kb.List2
+              items={items}
+              renderItem={renderItem}
+              itemHeight={{height: Styles.isMobile ? 48 : mode === 'self' ? 72 : 56, type: 'fixed'}}
+            />
+          </Kb.Box2>
         </Kb.Box2>
-      </Kb.Box2>
+      )}
     </Kb.Modal>
   )
 }
 
-const HeaderRow = ({onCreate, onSelectAll, onSelectNone}) => (
+const HeaderRow = ({mode, onCreate, onSelectAll, onSelectNone}) => (
   <Kb.Box2
     direction="horizontal"
     alignItems="center"
     fullWidth={true}
-    style={Styles.collapseStyles([styles.item, styles.headerItem])}
+    style={Styles.collapseStyles([styles.item, styles.headerItem, mode === 'self' && styles.itemSelf])}
   >
     <Kb.Button label="Create channel" small={true} mode="Secondary" onClick={onCreate} />
-    <Kb.Text type="BodyPrimaryLink" onClick={onSelectAll || onSelectNone}>
-      {onSelectAll ? 'Select all' : 'Clear'}
-    </Kb.Text>
+    {mode === 'self' ? (
```

Comment:
do we need a Box? `mode !== 'self' &&`

Reply:
It's for the left child to align itself properly with `space-between`. I added a comment.